internal/encrypt/crypt.go

```diff
@@ -0,0 +1,233 @@
+package encrypt
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/inconshreveable/log15"
+)
+
+const (
+	validKeyLength    = 32
+	hmacSize          = sha256.Size
+	primaryKeyIndex   = 0
+	secondaryKeyIndex = 1
+)
+
+type EncryptionError struct {
+	Message string
+	Err     error
+}
+
+func (e *EncryptionError) Error() string { return e.Message + ": " + e.Err.Error() }
+func (e *EncryptionError) Unwrap() error { return e.Err }
+
+// ErrNotEncoded means we can test for whether or not a string is encoded, prior to attempting decryption
+var ErrNotEncoded = errors.New("object is not encoded")
+
+// Generate a valid key for AES-256 encryption
+func GenerateRandomAESKey() ([]byte, error) {
+	b := make([]byte, validKeyLength)
+	_, err := rand.Read(b)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+// EncryptBytesIfPossible encrypts the byte array if encryption is configured.
+// Returns an error only when encryption is enabled, and encryption fails.
+func EncryptBytesIfPossible(b []byte) (string, error) {
+	if configuredToEncrypt {
+		return cryptObject.EncryptBytes(b)
+	}
+	return string(b), nil
+}
+
+// EncryptIfPossible encrypts the string if encryption is configured.
+// Returns an error only when encryption is enabled, and encryption fails.
+func EncryptIfPossible(value string) (string, error) {
+	if configuredToEncrypt {
+		return cryptObject.Encrypt(value)
+	}
+	return value, nil
+}
+
+// DecryptIfPossible decrypts the string if encryption is configured.
+// It returns an error only if encryption is enabled and it cannot Decrypt the string
+func DecryptIfPossible(value string) (string, error) {
+	if configuredToEncrypt {
+		return cryptObject.Decrypt(value)
+	}
+	return value, nil
+}
+
+// DecryptBytesIfPossible decrypts the byte array if encryption is configured.
+// It returns an error only if encryption is enabled and it cannot Decrypt the string
+func DecryptBytesIfPossible(b []byte) (string, error) {
+	if configuredToEncrypt {
+		return cryptObject.DecryptBytes(b)
+	}
+	return string(b), nil
+}
+
+// Encryptor contains the encryption key used in encryption and decryption.
+type Encryptor struct {
+	// the first key is always used to EncryptBytes, attempt to Decrypt with every key
+	EncryptionKeys [][]byte
+}
+
+// Returns an encrypted string.
+func (e *Encryptor) EncryptBytes(b []byte) (string, error) {
+	if len(e.EncryptionKeys) == 0 || e.EncryptionKeys[primaryKeyIndex] == nil {
+		return string(b), nil
+	}
+
+	// create a one time nonce of standard length, without repetitions
+	// ONLY use the primary key to EncryptBytes
+	block, err := aes.NewCipher(e.EncryptionKeys[primaryKeyIndex])
+	if err != nil {
+		return "", &EncryptionError{"cipher err:", err}
```

Comment:
```suggestion
		return "", &EncryptionError{"new cipher:", err}
```

Why this `returns` an `EncryptionError` but other `returns` in this method don't? Seems inconsistent.